### DISCUSSION TITLE
Let gpt-engineer ask for the prompt, instead of creating a prompt file

### DISCUSSION THREAD

**Discussion by joeytwiddle** [AntonOsika/gpt-engineer#604] [Planning]
## Motivation

As with AutoGPT, most of the work we do with `gpt-engineer` is done by typing responses to its questions. So why not make the _entire_ process like that?

Remove the need for me to create a new folder and edit the `prompt` file. Instead, let me simply run:

```
gpt-engineer my-new-project
or just
gpt-engineer
```

(We can keep this backward compatible with anyone who wants to continue to use the old process.)

## Current Behavior

I must copy `projects/example` to `projects/my_new_project` and edit the `prompt` file, before running `gpt-engineer`

If I had ever run `gpt-engineer` without any arguments before, it might have populated the `example` folder with work, so I need to clear those files.

## Suggested Behavior

Make it simpler: If I run `gpt-engineer` without a project argument, it will ask:

```
Please give the name of your new project?
```

And if there is no `prompt` file in that folder:

```
Please describe what you want the new project to do?
```

`gpt-engineer` will save the description into the `prompt` file, and then continue as usual.

**Comment by AntonOsika** [comment]
I've been thinking about optimal flow too.

Happy to get your input on reimagining it for the better.

# What I think we want to support
1. Run gpt-engineer to improve existing non-gpt-engineer generated codebases
2. Create a new codebase from scratch

# Proposed CLI flow
(command `ge`= short syntax for gpt-engineer)

Put all prompt files, memory etc in `.gpt_engineer` folder.

1. Run `ge`. Asks if you want to:
  a. Generate a new project (`create`)
  b. Update existing code (`fix`)
2. If new project, either:
  a. .gpt_engineer folder exists in current directory, if so, use prompt there.
  b. Ask for project name. If folder doesn't exist, create it. If prompt in project name folder/.gpt_engineer doesn't exist, create it. Before executing, allow user to edit prompt file that is generated. Try to execute code. Ask human if code worked; if not, try to fix and run again etc.
3. If updating existing code, ask user what files to include, and store choice to .gptengineer folder. Ask what should be fixed. Ask human if it worked, if not, try again etc.

## CLI options
Should be simple (and all be optional) and allow for:
1. Specifying working directory for project 
2. Specifying `fix` or `create`
3. Specifying parameters for what chain to use


Did I miss anything? Anything you don't like?

**Comment by joeytwiddle** [comment]
Glancing through this while on the move, some great ideas. My two cents:

- I would be happy to type my `prompt` directly into the REPL, as we do with the later questions. But file editing might make sense for some people with large specs.

  We could offer both with "Type your prompt or type 'e' to open an editor"

- Implementation-wise, you can manage a separate folder easily by doing `os.chdir(...)` early in the program, and then just working in the current folder. No need to pass the target folder all around the program.
